src/Components/CreateImageWizardV2/utilities/requestMapper.tsx

```diff
@@ -0,0 +1,164 @@
+import { RootState, store } from '../../../store';
+import {
+  AwsUploadRequestOptions,
+  CreateBlueprintRequest,
+  Customizations,
+  GcpUploadRequestOptions,
+  ImageRequest,
+  ImageTypes,
+  Subscription,
+  UploadTypes,
+} from '../../../store/imageBuilderApi';
+import {
+  selectActivationKey,
+  selectArchitecture,
+  selectAwsAccountId,
+  selectAwsShareMethod,
+  selectAwsSource,
+  selectBaseUrl,
+  selectBlueprintDescription,
+  selectBlueprintName,
+  selectCustomRepositories,
+  selectDistribution,
+  selectGcpAccountType,
+  selectGcpEmail,
+  selectGcpShareMethod,
+  selectImageTypes,
+  selectRegistrationType,
+  selectServerUrl,
+} from '../../../store/wizardSlice';
+
+/**
+ * This function maps the wizard state to a valid CreateBlueprint request object
+ * @param {string} orgID organization ID
+ * @returns {CreateBlueprintRequest} blueprint creation request payload
+ */
+export const mapRequestFromState = (orgID: string): CreateBlueprintRequest => {
+  const state = store.getState();
+  const imageRequests = getImageRequests(state);
+  const customizations = getCustomizations(state, orgID);
+
+  return {
+    name: selectBlueprintName(state),
+    description: selectBlueprintDescription(state),
+    distribution: selectDistribution(state),
+    image_requests: imageRequests,
+    customizations,
+  };
+};
+
+const getImageRequests = (state: RootState): ImageRequest[] => {
+  const imageTypes = selectImageTypes(state);
+  return imageTypes.map((type) => ({
+    architecture: selectArchitecture(state),
+    image_type: type,
+    upload_request: {
+      type: uploadTypeByTargetEnv(type),
+      options: getImageOptions(type, state),
+    },
+  }));
+};
+
+const uploadTypeByTargetEnv = (imageType: ImageTypes): UploadTypes => {
+  switch (imageType) {
+    case 'aws':
+      return 'aws';
+    case 'gcp':
+      return 'gcp';
+    case 'azure':
+      return 'azure';
+    case 'oci':
+      return 'oci.objectstorage';
+    case 'wsl':
+      return 'aws.s3';
+    case 'image-installer':
+      return 'aws.s3';
+    case 'vsphere':
+      return 'aws.s3';
+    case 'ami':
+      return 'aws';
+    default: {
+      // TODO: add edge type
+      throw new Error(`image type: ${imageType} has no implementation yet`);
+    }
+  }
+};
+const getImageOptions = (
+  imageType: ImageTypes,
+  state: RootState
+): AwsUploadRequestOptions | GcpUploadRequestOptions => {
+  switch (imageType) {
+    case 'aws':
+      if (selectAwsShareMethod(state) === 'sources')
+        return { share_with_sources: [selectAwsSource(state)?.id || ''] };
```

Comment:
I'm not sure about the `|| ''` here... in the state, `aws.source` can be either a `V1ListSourceResponseItem` or `undefined`. And the `id` field is optional on that type. So we do need to handle the case where there is not an id.

But I'm not sure passing an empty string is what we should do. After we build an AWS image, we upload it to an S3 bucket and share it with that AWS account via the id. If there is no id, then we won't be able to share the image, and it will be essentially inaccessible. 

So I think we probably ought to fail in that case. 

I need to think this through a bit more though.

@croissanne can I get a sanity check?

Reply:
yea for AWS if there's nothing to share it with, it's not accessible.

Reply:
Thanks @croissanne!

We will need to ensure that we check that there is an ID on the AWS step (I created an issue for that #1637).  

There are other edge cases to consider related to this but ultimately, we should hopefully never get to this point... if we do, it's a bug. Maybe it would be easier to track down the source of the bug if we just throw an error here, something like:
```typescript
throw (
  new Error('AWS account ID could not be resolved from source')
);
```

What do you think @amirfefer ?

Reply:
How about using `required` validation in the AWS step for the account number (if the sources radio is chosen)

<img width="982" alt="Screenshot 2024-02-08 at 20 18 02" src="https://github.com/osbuild/image-builder-frontend/assets/11807069/3da1db4d-aef4-4d57-b0af-836d0c4c279c">

I guess a  backend validation is needed as well?

Reply:
Yes, we'll definitely add that to the AWS step!

Reply:
But we still need to make Typescript happy and I don't think using an empty string is the right way... does throwing an error sound good to you? Like I mentioned below, I'd rather fail catastrophically and immediately rather than build an image that the user won't be able to access later...